onnxruntime/core/providers/webgpu/tensor/split.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/providers/webgpu/program.h"
+#include "core/providers/webgpu/webgpu_kernel.h"
+#include "core/providers/common.h"
+#include "core/providers/cpu/tensor/split.h"
+
+namespace onnxruntime {
+namespace webgpu {
+
+class SplitProgram final : public Program<SplitProgram> {
+ public:
+  SplitProgram(const uint32_t axis) : Program{"Split"}, axis_{axis} {}
+
+  Status GenerateShaderCode(ShaderHelper& sh) const override;
+
+  WEBGPU_PROGRAM_DEFINE_UNIFORM_VARIABLES({"input_size", ProgramUniformVariableDataType::Uint32},
+                                          {"sizes_in_split_axis", ProgramUniformVariableDataType::Uint32});
+
+ private:
+  uint32_t axis_;
+};
+
+class Split : public WebGpuKernel, public SplitBase {
+ public:
+  Split(const OpKernelInfo& info, uint32_t opset) : WebGpuKernel(info), SplitBase(info, opset) {
```

Comment:
Why do some validation in the `Split` constructor function, not in the `ComputeInternal` function? I see `PrepareForCompute` has already done some the same validation.

Reply:
If my understanding is right, in general the validation should be in the op's constructor function which is supposed to called during the session's initialization phase. For `Split` op, it can either have an attribute 'split' or input 'split'. If specified via input, its values are unavailable until the session's run phase, where the `ComputeInternal` function runs. This just follows what the CPU `Split` op does in the same [way](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/cpu/tensor/split.cc#L135).

Reply:
Just removed the validation in the constructor , which should be unnecessary as `PrepareForCompute ` basically does the same things.